orchestrator/db/models.py

```diff
@@ -200,18 +201,20 @@ class ProductTable(BaseModel):
     __table_args__ = {"extend_existing": True}
 
     def find_block_by_name(self, name: str) -> ProductBlockTable:
-        return (
-            object_session(self).query(ProductBlockTable).with_parent(self).filter(ProductBlockTable.name == name).one()
-        )
+        if session := object_session(self):
+            return session.query(ProductBlockTable).with_parent(self).filter(ProductBlockTable.name == name).one()
+        raise AssertionError("Session should not be None")
```

Comment:
Seeing as this occurs in 3 places, maybe add a function to the BaseModel class for this?

```
class BaseModel(_Base):


    ...

    def get_session(self) -> Session:
        if session := object_session(self):
            return session
        raise AssertionError("Session should not be None")
```

And use it like suggested
```suggestion
        return self.get_session().query(ProductBlockTable).with_parent(self).filter(ProductBlockTable.name == name).one()
```